noir/noir-repo/noir_stdlib/src/scalar_mul.nr

```diff
@@ -26,15 +27,33 @@ impl Add for EmbeddedCurvePoint {
 #[foreign(fixed_base_scalar_mul)]
 // docs:start:fixed_base_embedded_curve
 pub fn fixed_base_embedded_curve(
-    low: Field,
-    high: Field
+    low: Field, // low limb of the scalar
+    high: Field // high limb of the scalar
 ) -> [Field; 2]
 // docs:end:fixed_base_embedded_curve
 {}
 
+// Computes a variable base scalar multiplication over the embedded curve.
+// For bn254, We have Grumpkin and Baby JubJub.
+// For bls12-381, we have JubJub and Bandersnatch.
+//
+// The embedded curve being used is decided by the 
+// underlying proof system.
+// TODO(https://github.com/noir-lang/noir/issues/4931): use a point struct instead of two fields
+#[foreign(variable_base_scalar_mul)]
+// docs:start:variable_base_embedded_curve
+pub fn variable_base_embedded_curve(
+    point_x: Field, // x coordinate of a point to multiply the scalar with
+    point_y: Field, // y coordinate of a point to multiply the scalar with
+    scalar_low: Field, // low limb of the scalar
+    scalar_high: Field // high limb of the scalar
+) -> [Field; 2]
+// docs:end:variable_base_embedded_curve
+{}
+
 // This is a hack as returning an `EmbeddedCurvePoint` from a foreign function in brillig returns a [BrilligVariable::SingleAddr; 2] rather than BrilligVariable::BrilligArray
 // as is defined in the brillig bytecode format. This is a workaround which allows us to fix this without modifying the serialization format.
-fn embedded_curve_add(point1: EmbeddedCurvePoint, point2: EmbeddedCurvePoint) -> EmbeddedCurvePoint {
+pub fn embedded_curve_add(point1: EmbeddedCurvePoint, point2: EmbeddedCurvePoint) -> EmbeddedCurvePoint {
```

Comment:
Had to expose this to get `noir/noir-repo/test_programs/execution_success/variable_base_scalar_mul/src/main.nr` to pass testing brillig there.

Reply:
This is exposed already through the `Add` trait on `EmbeddedCurvePoint`